contrib/cirrus/lib.sh

```diff
@@ -288,3 +288,40 @@ remove_packaged_podman_files() {
     # Be super extra sure and careful vs performant and completely safe
     sync && echo 3 > /proc/sys/vm/drop_caches || true
 }
+
+# Execute make localbenchmarks in $CIRRUS_WORKING_DIR/data
+# for preserving as a task artifact.
+localbenchmarks() {
+    local datadir
+    req_env_vars DISTRO_NV PODBIN_NAME PRIV_NAME TEST_ENVIRON TEST_FLAVOR
+    req_env_vars VM_IMAGE_NAME EC2_INST_TYPE
+
+    datadir=$CIRRUS_WORKING_DIR/data
+    mkdir -p $datadir
+
+    # Most of these are defined by cirrus and it's YAML.
+    # shellcheck disable=SC2154,SC2153
+    cat <<EOF>$datadir/benchmarks.env
+CIRRUS_BASE_BRANCH=$CIRRUS_BASE_BRANCH
+CIRRUS_BRANCH=$CIRRUS_BRANCH
+CIRRUS_BUILD_ID=$CIRRUS_BUILD_ID
+CIRRUS_CHANGE_IN_REPO=$CIRRUS_CHANGE_IN_REPO
+CIRRUS_CRON=$CIRRUS_CRON
+CIRRUS_PR=$CIRRUS_PR
+CIRRUS_TASK_ID=$CIRRUS_TASK_ID
+DISTRO_NV=$DISTRO_NV
+PODBIN_NAME=$PODBIN_NAME
+PRIV_NAME=$PRIV_NAME
+TEST_ENVIRON=$TEST_ENVIRON
+TEST_FLAVOR=$TEST_FLAVOR
+VM_IMAGE_NAME=$VM_IMAGE_NAME
+INST_TYPE=$EC2_INST_TYPE
+UNAME_RM=$(uname -r -m)
+CPUTOTAL=$(echo $(($(egrep 'processor[[:space:]]+:' /proc/cpuinfo | tr -d '[:blank:]' | cut -d: -f2 | sort -n | tail -1) + 1)) )
+MEMTOTAL=$(grep 'MemTotal:' /proc/meminfo | tr -d '[:space:]' | cut -d: -f2)
```

Comment:
Might be more readable: `awk -F: '$1 == "MemTotal" { print $2 }' </proc/meminfo`